app/src/main/java/com/univoice/feature/signup/SchoolInputViewModel.kt

```diff
@@ -1,11 +1,32 @@
 package com.univoice.feature.signup
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.univoice.domain.repository.UniversityRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class SchoolInputViewModel : ViewModel() {
-    val mockSchoolList = listOf(
-        "서울대학교", "서울과학기술대학교", "서울시립대학교", "서울여자대학교", "서울학교5", "서울학교6",
-        "서울학교7", "서울학교8", "서울학교9", "서울학교10", "서울학교11", "서울학교12", "서울학교13",
-        "서울학교14", "서울학교15", "서울학교16", "서울학교17", "서울학교18", "서울학교19", "서울학교20", "서울학교21"
-    )
-}
\ No newline at end of file
+@HiltViewModel
+class SchoolInputViewModel @Inject constructor(
+    private val universityRepository: UniversityRepository
+) : ViewModel() {
+
+    private val _schoolList = MutableStateFlow<List<String>>(emptyList())
+    val schoolList: StateFlow<List<String>> = _schoolList
```

Comment:
P2 : UiState를 사용하면 API 통신할 때의 상태를 손쉽게 관리할 수 있어서 권장합니다!
```suggestion
    private val _postUniversityState = MutableStateFlow<UiState<List<String>>>(UiState.Empty)
    val postUniversityState: StateFlow<<UiState<List<String>> = _postUniversityState
```